### PR TITLE
[DDIA-308] Assign Service Instance in DNS Config Profile 

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: docs/resources/dns_host.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `id` (String) The resource identifier.
+
+### Optional
+
+- `associated_server` (Attributes) Host associated server configuration. (see [below for nested schema](#nestedatt--associated_server))
```
Comment (mathewab) [inline docs/resources/dns_host.md:52]: what is different in `associated_server` from `server`. I believe both are same, and one can be made readonly

Comment (unasra) [reply docs/resources/dns_host.md]: Made associated_server read only

File: docs/resources/dns_host.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+### Required
+
+- `id` (String) The resource identifier.
+
+### Optional
+
+- `inheritance_sources` (Attributes) Optional. Inheritance configuration. (see [below for nested schema](#nestedatt--inheritance_sources))
+- `retry_if_not_found` (Boolean) If set to `true`, the resource will retry until a matching host is found, or until the Create Timeout expires.
+- `server` (String) The resource identifier.
+- `tags` (Map of String) Host tagging specifics.
```
Comment (mathewab) [inline docs/resources/dns_host.md:55]: can tags be assigned to the host here? I believe it is read only.

File: docs/resources/dns_host.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+- `anycast_addresses` (List of String) Anycast address configured to the host. Order is not significant.
+- `associated_server` (Attributes) Host associated server configuration. (see [below for nested schema](#nestedatt--associated_server))
+- `comment` (String) Host description.
+- `current_version` (String) Host current version.
+- `dfp` (Boolean) Below _dfp_ field is deprecated and not supported anymore. The indication whether or not BloxOne DDI DNS and BloxOne TD DFP are both active on the host will be migrated into the new _dfp_service_ field.
+- `dfp_service` (String) DFP service indicates whether or not BloxOne DDI DNS and BloxOne TD DFP are both active on the host. If so, BloxOne DDI DNS will augment recursive queries and forward them to BloxOne TD DFP. Allowed values:  * _unavailable_: BloxOne TD DFP application is not available,  * _enabled_: BloxOne TD DFP application is available and enabled,  * _disabled_: BloxOne TD DFP application is available but disabled.
+- `kerberos_keys` (Attributes List) Optional. _kerberos_keys_ contains a list of keys for GSS-TSIG signed dynamic updates.  Defaults to empty. (see [below for nested schema](#nestedatt--kerberos_keys))
```
Comment (mathewab) [inline docs/resources/dns_host.md:68]: kerberos_keys can be assigned here, they shouldn't be read only

File: docs/resources/dns_host.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+- `inheritance_sources` (Attributes) Optional. Inheritance configuration. (see [below for nested schema](#nestedatt--inheritance_sources))
+- `retry_if_not_found` (Boolean) If set to `true`, the resource will retry until a matching host is found, or until the Create Timeout expires.
+- `server` (String) The resource identifier.
+- `tags` (Map of String) Host tagging specifics.
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+### Read-Only
+
+- `absolute_name` (String) Host FQDN.
```
Comment (mathewab) [inline docs/resources/dns_host.md:60]: absolute_name is not supposed to be read only and can be provided here.

Comment (unasra) [reply docs/resources/dns_host.md:60]: As per swagger , absolute name is readonly

File: docs/resources/dns_host.md (hunk 2 above)
Comment (unasra) [reply docs/resources/dns_host.md]: fixed

File: docs/resources/dns_host.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  }
+}
+
+resource "bloxone_dns_server" "dhcp_server" {
+  name = "my-dns-server"
+}
+
+resource "bloxone_dns_host" "example_dns_host" {
+  id     = data.bloxone_infra_hosts.host.results.0.legacy_id
+  server = bloxone_dns_server.dhcp_server.id
```
[the remark this answers is not in the document]
Comment (unasra) [reply docs/resources/dns_host.md]: fixed

File: docs/resources/dns_host.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+## Example Usage
+
+```terraform
+data "bloxone_infra_hosts" "host" {
+  filters = {
+    display_name = "my-host"
+  }
+}
+
+resource "bloxone_dns_server" "dhcp_server" {
```
[the remark this answers is not in the document]
Comment (unasra) [reply docs/resources/dns_host.md]: fixed

File: docs/resources/dns_host.md (hunk 4 above)
Comment (mathewab) [reply docs/resources/dns_host.md:59]: i see, however the UI and the API do allow passing values to it.

**Review by mathewab — APPROVED** [review]

**Pull request merged**